e3sm_diags/driver/zonal_mean_xy_driver.py

```diff
@@ -1,262 +1,287 @@
+        (e.g., one is 2-D and the other is 3-D).
+    """
     variables = parameter.variables
     seasons = parameter.seasons
     ref_name = getattr(parameter, "ref_name", "")
     regions = parameter.regions
 
-    test_data = utils.dataset.Dataset(parameter, test=True)
-    ref_data = utils.dataset.Dataset(parameter, ref=True)
-
-    for season in seasons:
-        # Get the name of the data, appended with the years averaged.
-        parameter.test_name_yrs = utils.general.get_name_and_yrs(
-            parameter, test_data, season
-        )
-        parameter.ref_name_yrs = utils.general.get_name_and_yrs(
-            parameter, ref_data, season
-        )
-
-        for var in variables:
-            logger.info("Variable: {}".format(var))
-            parameter.var_id = var
-
-            mv1 = test_data.get_climo_variable(var, season)
-            mv2 = ref_data.get_climo_variable(var, season)
-
-            parameter.viewer_descr[var] = (
-                mv1.long_name
-                if hasattr(mv1, "long_name")
-                else "No long_name attr in test data."
+    for region in regions:
+        if region != "global":
+            raise RuntimeError(
+                f"Region ({region}) is not supported. Only global region is currently "
+                "supported for the zonal_mean_xy set."
             )
 
-            # Special case, cdms didn't properly convert mask with fill value
-            # -999.0, filed issue with Denis.
-            if ref_name == "WARREN":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 == -0.9, mv2)
-            # The following should be moved to a derived variable.
-            if ref_name == "AIRS":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 > 1e20, mv2)
-            if ref_name == "WILLMOTT" or ref_name == "CLOUDSAT":
-                # This is cdms2 fix for bad mask, Denis' fix should fix this.
-                mv2 = MV2.masked_where(mv2 == -999.0, mv2)
-
-                # The following should be moved to a derived variable.
-                if var == "PRECT_LAND":
-                    days_season = {
-                        "ANN": 365,
-                        "DJF": 90,
-                        "MAM": 92,
-                        "JJA": 92,
-                        "SON": 91,
-                    }
-                    # mv1 = mv1 * days_season[season] * 0.1 # following AMWG
-                    # Approximate way to convert to seasonal cumulative
-                    # precipitation, need to have solution in derived variable,
-                    # unit convert from mm/day to cm.
-                    mv2 = (
-                        mv2 / days_season[season] / 0.1
-                    )  # Convert cm to mm/day instead.
-                    mv2.units = "mm/day"
```

Comment:
I noticed this entire section is removed without any new Xarray code to replace it. Is this block of code no longer needed since they are "cdms2" fixes?

I want to make sure because this chunk of code is copy-pasted in `zonal_mean_2d_driver.py`, which I am refactoring in #774.

https://github.com/E3SM-Project/e3sm_diags/blob/3e49671dd732b0221baf3e164097a9895088bd66/e3sm_diags/driver/zonal_mean_2d_driver.py#L113-L142

Reply:
Tom, I really appreciate the pointers you provided here and the fixes and updates to the PR. I'm resuming this work. 

For this question, this block of codes were useful at the very beginning of e3sm diags,( when its name was acme diags) for a set of legacy datasets inherited from AMWG package. They are no longer needed when we compile the new obs data inventory. They should have been removed earlier, and we can clean themn up during the migration.

Reply:
Of course! And thanks for confirming that we can deprecate this code.

This section is found in the polar driver as well. I'll leave a comment for @forsyth2 in #749 so that he is aware.